src/renderers.h

```diff
@@ -0,0 +1,85 @@
+/* renderers.h - header for renderers */
+
+/*
+* Copyright (c) 2017, Reddit
+*
+* Permission to use, copy, modify, and distribute this software for any
+* purpose with or without fee is hereby granted, provided that the above
+* copyright notice and this permission notice appear in all copies.
+*
+* THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+* WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+* MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+* ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+* WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+* ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+* OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+#include "../html/html.h"
+
+enum snudown_renderer_mode {
+	RENDERER_USERTEXT = 0,
+	RENDERER_WIKI,
+	RENDERER_USERTEXT_WITHOUTLINKS,
+	RENDERER_COUNT
+};
+
+struct snudown_renderopt {
+	struct html_renderopt html;
+	int nofollow;
+	const char *target;
+};
+
+struct snudown_renderer {
+	struct sd_markdown* main_renderer;
+	struct sd_markdown* toc_renderer;
+	struct module_state* state;
+	struct module_state* toc_state;
+};
+
+struct module_state {
+	struct sd_callbacks callbacks;
+	struct snudown_renderopt options;
+};
+
+static char* html_element_whitelist[] = { "tr", "th", "td", "table", "tbody", "thead", "tfoot", "caption", NULL };
```

Comment:
This is wrong. A `static` declaration in a header makes a copy of the thing in every compilation unit in which it is included.

Declare the variable as `extern` here, and define it in the renderers.c, so every reference uses the one compiled into renderers.o